test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctl
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/jetstack/cert-manager/cmd/ctl/pkg/create/certificaterequest"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+	"github.com/jetstack/cert-manager/test/integration/framework"
+)
+
+func TestCtlCreateCR(t *testing.T) {
+	config, stopFn := framework.RunControlPlane(t)
+	defer stopFn()
+
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Second*20)
+	defer cancel()
+
+	// Build clients
+	_, _, cmCl, _ := framework.NewClients(t, config)
+
+	// Create tmp directory and cd into it to store private key files
+	dir, err := ioutil.TempDir(".", "tmp")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chdir(dir); err != nil {
```

Comment:
Can we avoid calling `os.Chdir` altogether? It isn't safe to call within test cases as the next test case to run will be run starting in this named directory. From what I can tell, the 'current working directory' does not need to be set to anything special as we can use absolute paths as arguments for input and output?

Otherwise, if we _do_ need to set the current working directory, we could hide _all_ of this code behind something like:

```
func setupPathForTest(t *testing.T) func() {
    // call ioutil.TempDir here
    workingDirectoryBeforeTest := os.Workdir()
    // call os.Chdir here to change to the *new* working directory
    return func() {
      // call os.Chdir here to 'change' back to the *old* working directory
    }
}
```

This makes it nice and easy to use/read, with something like:

```
cleanupFunc := setupPathForTest(t)
defer cleanupFunc()
```

But as before, if we can avoid `os.Chdir` altogether that'd be best 😄